services/121-service/src/registration/repositories/registration-scoped.repository.spec.ts

```diff
@@ -0,0 +1,282 @@
+import { Test } from '@nestjs/testing';
+import { DataSource } from 'typeorm';
+
+import { RegistrationScopedRepository } from '@121-service/src/registration/repositories/registration-scoped.repository';
+import { ScopedUserRequest } from '@121-service/src/shared/scoped-user-request';
+
+describe('RegistrationScopedRepository - getFullNamesByRegistrationIds', () => {
+  let repository: RegistrationScopedRepository;
+  let mockQueryBuilder: any;
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+
+    mockQueryBuilder = {
+      select: jest.fn().mockReturnThis(),
+      addSelect: jest.fn().mockReturnThis(),
+      innerJoin: jest.fn().mockReturnThis(),
+      leftJoin: jest.fn().mockReturnThis(),
+      andWhere: jest.fn().mockReturnThis(),
+      getRawMany: jest.fn().mockResolvedValue([]),
+      groupBy: jest.fn().mockReturnThis(),
+      orderBy: jest.fn().mockReturnThis(),
+      limit: jest.fn().mockReturnThis(),
+    };
+
+    const mockRepository = {
+      createQueryBuilder: jest.fn().mockReturnValue(mockQueryBuilder),
+    };
+
+    const mockEntityManager = {
+      getRepository: jest.fn().mockReturnValue(mockRepository),
+    };
+
+    const mockDataSource = {
+      createEntityManager: jest.fn().mockReturnValue(mockEntityManager),
+    };
+
+    const module = await Test.createTestingModule({
+      providers: [
+        RegistrationScopedRepository,
+        {
+          provide: DataSource,
+          useValue: mockDataSource,
+        },
+        {
+          provide: 'REQUEST',
+          useValue: {} as ScopedUserRequest,
+        },
+      ],
+    }).compile();
+
+    // Use resolve() instead of get() because that does not work for request-scoped providers
+    repository = await module.resolve<RegistrationScopedRepository>(
+      RegistrationScopedRepository,
+    );
+
+    repository.createQueryBuilder = jest.fn().mockImplementation(() => {
+      return mockQueryBuilder;
+    });
+  });
+
+  it('should return empty names when fullNameNamingConvention is empty', async () => {
```

Comment:
Removing the concept of fullNameNamingConvention would make this code a lot shorter :)
If not now, then let's do it as part of the Kobo Toolbox integration work.